Compiling OpenCV3.4.1 with Java module failed (attached workaround)

<!--
If you have a question rather than reporting a bug please go to http://answers.opencv.org where you get much faster responses.
If you need further assistance please read [How To Contribute](https://github.com/opencv/opencv/wiki/How_to_contribute).

This is a template helping you to create an issue which can be processed as quickly as possible. This is the bug reporting section for the OpenCV library.
-->

##### System information (version)
<!-- Example
- OpenCV => 3.1
- Operating System / Platform => Windows 64 Bit
- Compiler => Visual Studio 2015
-->
- OpenCV => 3.4.1
- Operating System / Platform => CentOS 7 64 Bit
- Compiler => GCC 4.8.5

##### Detailed description

<!-- your description -->

Compiling OpenCV 3.4.1 from source with Java module

Occurred:
Traceback (most recent call last):
  File "/root/opencv/sources/modules/java/generator/../generator/gen_java.py", line 1094, in <module>
    copy_java_files(java_files_dir, target_path)
  File "/root/opencv/sources/modules/java/generator/../generator/gen_java.py", line 1043, in copy_java_files
    package_path = 'org/opencv/' + module
NameError: global name 'module' is not defined
make[2]: *** [CMakeFiles/dephelper/gen_opencv_java_source] Error 1
make[1]: *** [modules/java_bindings_generator/CMakeFiles/gen_opencv_java_source.dir/all] Error 2
make: *** [all] Error 2

Workaround:
edit /root/opencv/sources/modules/java/generator/../generator/gen_java.py
change line 1029 from:
re_package = re.compile(r'^package +(.+);$')
to
re_package = re.compile(r'^package +(.+);')

<!-- to add code example fence it with triple backticks and optional file extension
    ```.cpp
    // C++ code example
    ```
 or attach as .txt or .zip file
-->

@alalek It is the result of adding the lines you mentioned:

> /home/bahramudin-adil/Desktop/opencv/opencv/modules/java/generator/src/java/org/opencv/utils/Converters.java
> 'package org.opencv.utils;\r\n'
> Traceback (most recent call last):
>   File "/home/bahramudin-adil/Desktop/opencv/opencv/modules/java/generator/../generator/gen_java.py", line 1095, in <module>
>     copy_java_files(java_files_dir, target_path)
>   File "/home/bahramudin-adil/Desktop/opencv/opencv/modules/java/generator/../generator/gen_java.py", line 1044, in copy_java_files
>     package_path = 'org/opencv/' + module
> NameError: global name 'module' is not defined
> modules/java_bindings_generator/CMakeFiles/gen_opencv_java_source.dir/build.make:430: recipe for target 'CMakeFiles/dephelper/gen_opencv_java_source' failed
> make[2]: *** [CMakeFiles/dephelper/gen_opencv_java_source] Error 1
> CMakeFiles/Makefile2:2661: recipe for target 'modules/java_bindings_generator/CMakeFiles/gen_opencv_java_source.dir/all' failed
> make[1]: *** [modules/java_bindings_generator/CMakeFiles/gen_opencv_java_source.dir/all] Error 2
> make[1]: *** Waiting for unfinished jobs....

--- a/modules/java/generator/gen_java.py
+++ b/modules/java/generator/gen_java.py
@@ -1018,7 +1018,7 @@ JNIEXPORT void JNICALL Java_org_opencv_%(module)s_%(j_cls)s_delete
         self.save(list_file, '\n'.join(['#include "%s"' % f for f in self.cpp_files]))
 
 
-def copy_java_files(java_files_dir, java_base_path):
+def copy_java_files(java_files_dir, java_base_path, default_package_path='org/opencv/'):
     global total_files, updated_files
     java_files = []
     re_filter = re.compile(r'^.+\.(java|aidl)(.in)?$')
@@ -1026,7 +1026,7 @@ def copy_java_files(java_files_dir, java_base_path):
        java_files += [os.path.join(root, filename) for filename in filenames if re_filter.match(filename)]
     java_files = [f.replace('\\', '/') for f in java_files]
 
-    re_package = re.compile(r'^package +(.+);$')
+    re_package = re.compile(r'^package +(.+);')
     re_prefix = re.compile(r'^.+[\+/]([^\+]+).(java|aidl)(.in)?$')
     for java_file in java_files:
         src = checkFileRemap(java_file)
@@ -1039,7 +1039,7 @@ def copy_java_files(java_files_dir, java_base_path):
             package = m.group(1)
             package_path = package.replace('.', '/')
         else:
-            package_path = 'org/opencv/' + module
+            package_path = default_package_path
         #print(java_file, package_path, target_fname)
         dest = os.path.join(java_base_path, os.path.join(package_path, target_fname))
         assert dest[-3:] != '.in', dest + ' | ' + target_fname
@@ -1154,11 +1154,11 @@ if __name__ == "__main__":
 
         java_files_dir = os.path.join(misc_location, 'src/java')
         if os.path.exists(java_files_dir):
-            copy_java_files(java_files_dir, java_base_path)
+            copy_java_files(java_files_dir, java_base_path, 'org/opencv/' + module)
 
         java_test_files_dir = os.path.join(misc_location, 'test')
         if os.path.exists(java_test_files_dir):
-            copy_java_files(java_test_files_dir, java_test_base_path)
+            copy_java_files(java_test_files_dir, java_test_base_path, 'org/opencv/test/' + module)
 
         if len(srcfiles) > 0:
             generator.gen(srcfiles, module, dstdir, jni_path, java_path, common_headers)

